src/main/scala/io/iohk/ethereum/domain/Address.scala

```diff
@@ -0,0 +1,28 @@
+package io.iohk.ethereum.domain
+
+import akka.util.ByteString
+import io.iohk.ethereum.vm.DataWord
+
+case object Address {
+
+  val Length = 20
+
+  def apply(dw: DataWord): Address = Address(dw.bytes.takeRight(Length))
+
+  def apply(arr: Array[Byte]): Address = Address(ByteString(arr))
+
+  val empty: Address = Address(ByteString())
+
+}
+
+case class Address(bytes: ByteString) {
```

Comment:
It has been pointed out about using `IndexedSeq[Byte]` ... @whysoserious do you think we can use it here too?